lib/core/Head.js

```diff
@@ -16,13 +16,19 @@ class Head extends React.Component {
       if (link.blog) hasBlog = true;
     });
 
-    const highlightDefaultVersion = '9.12.0';
-    const highlightConfig = this.props.config.highlight || {
-      version: highlightDefaultVersion,
+    let highlight = {
+      version: '9.12.0',
       theme: 'default',
+      ...this.props.config.highlight,
     };
-    const highlightVersion = highlightConfig.version || highlightDefaultVersion;
-    const highlightTheme = highlightConfig.theme || 'default';
+
+    if (highlight.theme.indexOf('/') > -1) {
+      highlight = highlight.theme;
```

Comment:
It's not a good idea to let the theme be either a string or a URL as this leads to confusing configuration. It's better to add a new field in `highlight` object called `themeUrl` and the code here uses it directly instead of building up the URL from the version and theme.

Define a new variable `highlightThemeURL` instead. Don't reassign a value of a different type to the `highlight` variable.

```js
// Use user-provided themeUrl if it exists, else construct one from version and theme.
let highlightThemeURL = highlight.themeUrl || `//cdnjs.cloudflare.com/ajax/libs/highlight.js/${highlight.version}/styles/${highlight.theme}.min.css`;

// Use highlightThemeURL in the `<link>`.
```